Observing a Python path Issue with Git based Compile

**Describe the bug**
environment variables are sourced but Python version is different when/with cmake runs

**To Reproduce**
Steps to reproduce the behavior:
using Gitbash to set environment and start build.
 
**$ cmake -GNinja -DBOARD=stm32_min_dev**
_CMake Error at C:/ProgApps/cmake-3.13.1-win64-x64/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find PythonInterp: Found unsuitable version "1.4", but required_
  is at least "3.4" (found C:/ProgApps/Anaconda/python.exe)
Call Stack (most recent call first):
  C:/ProgApps/cmake-3.13.1-win64-x64/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:376 (_FPHSA_FAILURE_MESSAGE)
  C:/ProgApps/Zephyr/cmake/backports/FindPythonInterp.cmake:165 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  C:/ProgApps/Zephyr/cmake/app/boilerplate.cmake:95 (include)
  CMakeLists.txt:3 (include)

-- Configuring incomplete, errors occurred!

**$ bash --version**
GNU bash, version 4.4.19(2)-release (x86_64-pc-msys)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

**$ where python**
C:\ProgApps\**Py37**\python.exe


**Expected behavior**
Able to compile and generate binary for STM32 similar to https://docs.zephyrproject.org/latest/boards/arm/stm32_min_dev/doc/stm32_min_dev.html

**Impact**
not able to compile on windows.

**Screenshots or console output**

**Environment (please complete the following information):**
 - OS: Windows 10 Enterprise
 - Toolchain 
VERSION_MAJOR = 1
VERSION_MINOR = 13
PATCHLEVEL = 99
VERSION_TWEAK = 0
EXTRAVERSION = d62e861a 

**Additional context**
Unable to configure the Python environment for Build to complete.
![gitbasherror](https://user-images.githubusercontent.com/4569591/49938156-08a5cf80-feff-11e8-8d77-0e1393a38e
[samples-hello_world.zip](https://github.com/zephyrproject-rtos/zephyr/files/2675927/samples-hello_world.zip)
c4.png)

Hi,
  I still see the Error after running Zephyr-env.cmd

C:\ProgApps\Zephyr\samples\hello_world>cmake -GNinja -DBOARD=stm32_min_dev
CMake Error at C:/ProgApps/cmake-3.13.1-win64-x64/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find PythonInterp: Found unsuitable version "1.4", but required
  is at least "3.4" (found C:/ProgApps/Anaconda/python.exe)
Call Stack (most recent call first):
  C:/ProgApps/cmake-3.13.1-win64-x64/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:376 (_FPHSA_FAILURE_MESSAGE)
  C:/ProgApps/Zephyr/cmake/backports/FindPythonInterp.cmake:165 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  C:/ProgApps/Zephyr/cmake/app/boilerplate.cmake:95 (include)
  CMakeLists.txt:3 (include)


-- Configuring incomplete, errors occurred!

I ran the same with Qemu [..](https://docs.zephyrproject.org/1.13.0/boards/x86/qemu_x86/doc/board.html)
 Directory of C:\ProgApps\Zephyr\samples\synchronization

12/06/2018  04:47 PM    <DIR>          .
12/06/2018  04:47 PM    <DIR>          ..
12/06/2018  04:47 PM               175 CMakeLists.txt
12/06/2018  04:47 PM                75 prj.conf
12/06/2018  04:47 PM             1,015 README.rst
12/06/2018  04:47 PM               696 sample.yaml
12/06/2018  04:47 PM    <DIR>          src
               4 File(s)          1,961 bytes
               3 Dir(s)  435,901,440,000 bytes free

C:\ProgApps\Zephyr\samples\synchronization>mkdir build

C:\ProgApps\Zephyr\samples\synchronization>cmake -GNinja -DBOARD=qemu_x86
-- Found PythonInterp: C:/ProgApps/Py37/python.exe (found suitable version "3.7.1", minimum required is "3.4")
CMake Error at C:/ProgApps/Zephyr/cmake/app/boilerplate.cmake:102 (message):
  Source directory equals build directory.  In-source builds are not
  supported.  Please specify a build directory, e.g.  cmake -Bbuild -H.
Call Stack (most recent call first):
  CMakeLists.txt:3 (include)


-- Configuring incomplete, errors occurred!

C:\ProgApps\Zephyr\samples\synchronization>where python
C:\ProgApps\Py37\python.exe

--- a/cmake/backports/FindPythonInterp.cmake
+++ b/cmake/backports/FindPythonInterp.cmake
@@ -43,11 +43,22 @@ function(determine_python_version exe result)
                     RESULT_VARIABLE _PYTHON_VERSION_RESULT
                     ERROR_QUIET)
     if(_PYTHON_VERSION_RESULT)
-        message(FATAL_ERROR "Failed to test python version")
+        # sys.version predates sys.version_info, so use that
+        execute_process(COMMAND "${exe}" -c "import sys; sys.stdout.write(sys.version)"
+                        OUTPUT_VARIABLE _VERSION
+                        RESULT_VARIABLE _PYTHON_VERSION_RESULT
+                        ERROR_QUIET)
+        if(_PYTHON_VERSION_RESULT)
+            # sys.version was first documented for Python 1.5, so assume
+            # this is older.
+            set(ver "1.4")
+        else()
+            string(REGEX REPLACE " .*" "" ver "${_VERSION}")
+        endif()
     else()
         string(REPLACE ";" "." ver "${_VERSION}")
-        set(${result} ${ver} PARENT_SCOPE)
     endif()
+    set(${result} ${ver} PARENT_SCOPE)
 endfunction()
 
 # Find out if the 'python' executable on path has the correct version,

